### PR TITLE
refactor: replacing react components

### PR DESCRIPTION
**Pull request by koonpeng**
## What's new

This massive PR refactor and migrates the current dashboard components to `react-components`. The following components are migrated

* All schedule visualizer markers
* Most omnipanel accordions (task and negotiations are not ported)
* Alert dialog
* Login page
* Some utils functions

These features are removed during the migration:

* "Outline" trajectory
* Settings for trajectory diameter
* Settings for trajectory color

Personally I don't think there is any demands/requirements for these features, but I can add them back if you think they are important.

## Self-checks

- [x] I'm familiar with and follow this [ Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [x] I added unit-tests for new components
- [x] I tried testing edge cases
- [x] I tested the behavior of the components that interact with the backend, with an e2e test

## Discussion

[Questions for reviewers]

### CODE REVIEW

**Review by matiasbavera — COMMENTED** [review]
Thanks for these changes. My browser crashed a couple of times doing this review on GH =)

Left some comments, I found this doing some testing
Lift: 
image

the path is thinner than before right?

image

Maybe it's just my but computer but now the dashboard is working with a lot of delay with `start:mock` 
slow

HTH

File: packages/dashboard/src/components/schedule-visualizer/dispensers-overlay.tsx [hunk 1]
```diff
@@ -32,19 +31,22 @@ export const DispensersOverlay = React.memo(
...
-                <Dispenser
+                <DispenserMarker
                   key={dispenser.guid}
-                  dispenser={dispenser}
-                  dispenserHandler={dispenserResourcesContext}
+                  guid={dispenser.guid}
+                  location={[dispenser.location.x, dispenser.location.y]}
+                  iconPath={dispenserResourcesContext.getIconPath(dispenser.guid) || undefined}
                   footprint={footprint}
-                  dispenserState={dispenserStates[dispenser.guid]}
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/src/components/schedule-visualizer/dispensers-overlay.tsx:46]: Good catch, when the dispenser panel is written, we do not have a resource file and rmf does not have a database of dispensers that we can query. So we end up with sort of a hack and only show dispensers that we received a state for, we probably should use the dispenser data from the resource file now.

File: packages/dashboard/src/components/app.tsx [hunk 2]
```diff
@@ -11,6 +12,16 @@ import { User } from './auth/user';
 import Dashboard from './dashboard';
 import NotFoundPage from './error-pages/page-not-found';
 
+const theme = createMuiTheme({
+  palette: {
+    primary: {
+      main: '#44497a',
+      dark: '#323558',
+      light: '#565d99',
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/src/components/app.tsx:20]: I opt for overriding the default theme because we want the material components to match the color scheme.

File: packages/dashboard/e2e/tests/utils.ts [hunk 3]
```diff
@@ -79,23 +86,23 @@ interface E2eRequestLoopProps {
...
     removeTextFromAutocomplete(charsToRemoveFromPointB || 20),
   );
-  $('input[name=finishLocation]').setValue(pointB);
+  $('input[placeholder="Pick Finish Location"]').setValue(pointB);
   $('.MuiAutocomplete-popper').click();
 
-  $('input[name=numLoops]').waitForClickable();
-  $('input[name=numLoops]').setValue(loopNumber || 1);
+  $('input[placeholder="Number of loops"]').waitForClickable();
+  $('input[placeholder="Number of loops"]').setValue(loopNumber || 1);
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/e2e/tests/utils.ts:106]: I think we can do it in another PR if the need arises.

**Comment by koonpeng** [comment]
> the path is thinner than before right?

Yes, the previous path is using a hardcoded width, we are now properly using the robot's dimension.

> Maybe it's just my but computer but now the dashboard is working with a lot of delay with start:mock

Thanks for the heads up, I will look into it.

**Comment by koonpeng** [comment]
I added some react memos, see if it works better now.

**Comment by matiasbavera** [comment]
> Yes, the previous path is using a hardcoded width, we are now properly using the robot's dimension. Actually the robot icons are too big because we are using a hardcoded value there, created an issue to track this as well.
> 
> >

Awesome!

**Comment by matiasbavera** [comment]
Having these warnings
 
```
./src/components/dashboard.tsx
  Line 50:10:  'DashboardTour' is defined but never used       @typescript-eslint/no-unused-vars
  Line 425:9:  'tourProps' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

It should be resolved if you rebase this branch with master

**Comment by matiasbavera** [comment]
Storybook `simpleData` it's failing

image

**Comment by matiasbavera** [comment]
Is intentional Having those lifts with the same color, even with different states?

image

**Review by matiasbavera — APPROVED** [review]
LGTM, did some testing, left a couple of comments. I think it's ready to merge when those are solved. I'm approving it to unblock the merge after the fixes.

HTH,

**Comment by matiasbavera** [comment]
> Storybook `simpleData` it's failing
> 
> image

Fixed

**Comment by matiasbavera** [comment]
Rebasing this PR
